api/v1alpha1/openstackrouter_types.go

```diff
@@ -0,0 +1,177 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// OpenStackRouterSpec defines the desired state of OpenStackRouter
+type OpenStackRouterSpec struct {
+	// Cloud is the OpenStackCloud hosting this resource
+	Cloud string `json:"cloud"`
+
+	// ID is the OpenStack UUID of the resource. If left empty, the
+	// controller will create a new resource and populate this field. If
+	// manually populated, the controller will adopt the corresponding
+	// resource.
+	ID string `json:"id,omitempty"`
+
+	// Name of the OpenStack resource.
+	Name string `json:"name,omitempty"`
+
+	Description string `json:"description,omitempty"`
+
+	ExternalGateway string `json:"externalGateway,omitempty"`
+
+	// AdminStateUp is the administrative state of the router.
+	AdminStateUp *bool `json:"adminStateUp,omitempty"`
```

Comment:
The API [docs](https://docs.openstack.org/api-ref/network/v2/#create-router) declares the default of this field to be `true`. Based on this, we can remove the pointer and make the value default to `true`:

```suggestion
	// +kubebuilder:default:=true
	AdminStateUp bool `json:"adminStateUp,omitempty"`
```

Reply:
The thing is, defaults sometimes change. Maybe your pointer solution was more future-proof. What do you think?

Reply:
After some thought, your pointer solution was more consistent with the API. Like the API, it lets you pick one of three choices: true, false or "you pick". Can you please revert to pointer and excuse me for the back-and-forth?

Reply:
Sure! no worries